Setup

theovincent opened the pull request:
I thought about setting a working environment with which everyone could work. I think it will prevent us from having problems with dependencies.

I would be happy to ear what you think about it. The main idea is to develop this repository as if it would be a package. I explain the step to follow in the README to install it.

emasquil commented:
This is a good idea!

I think maybe we can use a `requirements.txt` file and read it from the setup (like this for example https://stackoverflow.com/questions/26900328/install-dependencies-from-setup-py). But it's just a minor thing and we can do this change in the future if really needed (if we have lots of requirements)

paulinesert reviewed (commented):
@theovincent  @nviolante25 I think we need to add:

dm-acme==0.2.4
dm-control== 0.0.364896371
dm-env==1.5
dm-tree==0.1.6


Those were installed when i tried to run the env-wrapper code in the ppo venv 
I installed it running the following command : pip install dm-acme[envs]

I am not too sure how this can be translated for the setup.py so I have bluntly put the version / lib it has installed :) 

Also is it okay if any of us writes PR for the setup file? :)

nviolante25 commented on setup.py:
```diff
@@ -0,0 +1,14 @@
+    name="ppo",
+    version="0.1",
+    description="Simple PPO implementation on OpenAI GYM environment.",
+    packages=["src"],
```
Since we changed  `src` to `ppo`, shouldn't this be changed as well?

theovincent replied:
Yes, definitely. I am going to do that.

theovincent commented:
@paulinesert I am going to rebase this PR first and I will see which dependency we need to install!

theovincent commented:
I tried to respect your three requests:
- @emasquil the requirements are now in setup.cfg.
- @nviolante25  the package is now searched at `ppo`.
- @paulinesert  I have added `dm-acme` to the requirements. I tried to keep as less requirements as possible.

I deleted my personal virtual environment and build it again. It works fine for me!

ps: I rebased this branch on `main`. It is not the first time I try to rebase a branch but it is the first time I succeed haha. This is why I wrote a wiki [here](https://github.com/emasquil/ppo/wiki/Rebase) in case you need it :)

nviolante25 commented:
Great! 
A small detail about the link on the wiki about rebasing. Instead of using `--force`, it's generally better to use `--force-with-lease`. That prevents overwriting other people's work.

theovincent commented:
> Great! A small detail about the link on the wiki about rebasing. Instead of using `--force`, it's generally better to use `--force-with-lease`. That prevents overwriting other people's work.

Thank you for the comment. Maybe you can add it to the wiki?

nviolante25 commented:
> Thank you for the comment. Maybe you can add it to the wiki?

Sure!
